Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions-details.tsx

```diff
@@ -89,6 +96,50 @@ const SubmissionsDetailsProvider = ({ children }: ISubmissionsDetailsProviderPro
         parameters: submissionResultsByProblemUrlParams,
     });
 
+    const {
+        get: downloadSubmissionFile,
+        response: downloadSubmissionFileResponse,
+        error: downloadSubmissionFileError,
+        saveAttachment,
+    } = useHttp<IDownloadSubmissionFileUrlParams, Blob>({
+        url: getSubmissionFileDownloadUrl,
+        parameters: { id: problemSubmissionFileIdToDownload },
+    });
+
+    const downloadProblemSubmissionFile = useCallback(
+        async (submissionId: number) => {
+            setProblemSubmissionFileIdToDownload(submissionId);
+        },
+        [],
+    );
+
+    useEffect(() => {
+        if (isNil(downloadSubmissionFileResponse)) {
+            return;
+        }
+
+        if (!isNil(downloadSubmissionFileError)) {
+            setDownloadErrorMessage(downloadSubmissionFileError?.detail);
+            return;
+        }
+
+        saveAttachment();
+    }, [ downloadSubmissionFileResponse, saveAttachment, downloadSubmissionFileError ]);
+
+    useEffect(() => {
+        if (isNil(problemSubmissionFileIdToDownload)) {
+            return;
+        }
+
+        (async () => {
+            startLoading();
+            await downloadSubmissionFile(FileType.Blob);
```

Comment:
Why is it needed to pass the FileType to a download file function?

Reply:
I need to tell it that the respones type will be a blob, otherwise it sets it to json and it does not work.